### DISCUSSION TITLE
Allow numpy integers and numpy arrays of length 2 as inputs when integers and tuples resp. are required as inputs

### DISCUSSION THREAD

**Discussion by jasm37** [kornia/kornia#1556] [Potential Issue] [answered]
I was playing around with `kornia.filters.unsharp_mask` and noticed that only types `Tuple[int, int]` are accepted as possible parameters. I found out about this when I generated a sequence of tuples using `np.linspace(min_val, max_val, num_steps + 1, dtype=int)` to visualize how each `kernel_size` changes the output image. At first I got the following error:
```
/opt/conda/lib/python3.7/site-packages/kornia/filters/kernels.py in get_gaussian_kernel2d(kernel_size, sigma, force_even)
    451     """
    452     if not isinstance(kernel_size, tuple) or len(kernel_size) != 2:
--> 453         raise TypeError("kernel_size must be a tuple of length two. Got {}".format(kernel_size))
    454     if not isinstance(sigma, tuple) or len(sigma) != 2:
    455         raise TypeError("sigma must be a tuple of length two. Got {}".format(sigma))

TypeError: kernel_size must be a tuple of length two. Got [5 5]
``` 
where it complained that the two-element numpy array is not accepted. Then I casted each array into a tuple but then I got
```
/opt/conda/lib/python3.7/site-packages/kornia/filters/kernels.py in get_gaussian_kernel1d(kernel_size, sigma, force_even)
    357     """
    358     if not isinstance(kernel_size, int) or ((kernel_size % 2 == 0) and not force_even) or (kernel_size <= 0):
--> 359         raise TypeError("kernel_size must be an odd positive integer. " "Got {}".format(kernel_size))
    360     window_1d: torch.Tensor = gaussian(kernel_size, sigma)
    361     return window_1d

TypeError: kernel_size must be an odd positive integer. Got 5
```
where essentially now the entries in each tuple are of type `np.int32`, not `int`.

So summarizing I think these assertions should be generalized to include numpy ints and sequences of length 2 instead of only `int` and `tuple`. I guess this might apply to other filters but I have not try it out. Now this might not be important but can cause unexpected bugs and/or unpleasant code fixes for numpy arrays in the future. I myself was a bit confused at first when it happened

### For the first exception:
```
import numpy as np
import kornia as kor
img = np.zeros((3, 100, 100))
variable = np.array([5, 5], dtype=int) # or np.linspace(min_val, max_val, num_steps + 1, dtype=int)[0]
kor.filters.unsharp_mask(input=img, kernel_size=variable, sigma=(2, 2))
```
### For the second exception:
```
import numpy as np
import kornia as kor
img = np.zeros((3, 100, 100))
variable = tuple(np.array([5, 5], dtype=int))
kor.filters.unsharp_mask(input=img, kernel_size=variable, sigma=(2, 2))
```

**Comment by edgarriba** [comment, marked as answer]
Hi, Kornia it’s a pytorch library. Even having similar semantics, don’t expect to work well with numpy arrays. For the parameters, we usually assume primitive types unless it’s a potential learnable variable.

**Reply by jasm37** [reply]
Gotcha! I just wanted to point it out because I was using the library for educational reasons by sampling different parameters and plotting pretty pictures, but I was not expecting this behaviour. Thank you for the clarification!
